### PR TITLE
fix: Implemented the logic to force the user to tag at least one repositor…

### LINKED ISSUE #1995: Bug: Blog post highlights make up random repos
### Describe the bug

Screen Shot 2023-10-24 at 3 54 11 PM

### Steps to reproduce

The need for recognition is not a repo and every blog highlight should be linked to a repo. 

The solution should enforce a repo being tagged on a blog highlight.    

### Browsers

Chrome

### Additional context (Is this in dev or production?)

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by shamimbinnur**
## Description
This PR adds the logic to force the user to tag at least one repository if the system detects an empty array of repositories while creating a highlight.

The system has the logic to automatically detect repositories from the 'PR' and 'Issue' links and tag them to a highlight. However, for the dev.to blog, the system couldn't detect any related repositories, resulting in an empty array of repositories being passed to the backend. This is why an invalid repository was tagged from the backend.

To fix this issue, I have added logic to require users to manually tag at least one repository if the system detects an empty array of repositories while creating a highlight.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Fixes #1995 

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: components/molecules/HighlightInput/highlight-input-form.tsx [hunk 1]
```diff
@@ -432,6 +432,12 @@ const HighlightInputForm = ({ refreshCallback }: HighlightInputFormProps): JSX.E
 
       const taggedRepoFullNames = taggedRepoList.map((repo) => `${repo.repoOwner}/${repo.repoName}`);
 
+      // Check if the user has tagged at least one repo and ask them to tag at least one if they haven't
+      if (taggedRepoList.length < 1) {
+        setError("Please tag at least one repo!");
```
Comment (nickytonline) [inline components/molecules/HighlightInput/highlight-input-form.tsx:437]: wording (blocking): Make it clear why at least one repository needs to be added.

```suggestion
        setError("Please add at least one repository associated with the blog post");
```

@BekahHW, any suggestions for better wording?

**Review by nickytonline — COMMENTED** [review]
Thanks for the fix @shamimbinnur! We're just sorting out the wording and this should be good to merge.

**Review by OgDev-01 — APPROVED** [review]
Copy changes and we are good 🍕. Thanks for turning this around

File: components/molecules/HighlightInput/highlight-input-form.tsx (hunk 1 above)
Comment (BekahHW) [reply components/molecules/HighlightInput/highlight-input-form.tsx:437]: Ok, I think this is tricky. I need some clarification first. The problem is that when someone submits a blog post, they are automatically generating a repo, right?

I think the repo should be optional in for blog post highlights. For example, it's still valid to highlight work on something like "A guide to writing documentation." There's not necessarily a repo associated with that. 

Can we conditionally check if the highlight is a blog post, and then leave the input empty unless they put something in there?

Comment (BekahHW) [reply components/molecules/HighlightInput/highlight-input-form.tsx:437]: @bdougie thoughts on the above?

Comment (BekahHW) [reply components/molecules/HighlightInput/highlight-input-form.tsx:437]: If we move forward with this implementation, I think Nick's copy works just change the to your:

> "Please add at least one repository associated with your blog post"

Comment (nickytonline) [reply components/molecules/HighlightInput/highlight-input-form.tsx:437]: Good point that it needs to be when a blog post is added @BekahHW. Missed that. 😅 

@shamimbinnur, we'd need to check that it's a blog post before showing that error.

Comment (shamimbinnur) [reply components/molecules/HighlightInput/highlight-input-form.tsx:437]: Hey @nickytonline, thanks for suggesting the better wording.

Not allowing the user to post a blog without tagging the repo is intentional. As @bdougie's comment [here](https://github.com/open-sauced/app/issues/1995#issue-1959992628) states, "every blog highlight should be linked to a repo."

If no repo is tagged whether automatically or manually an empty array of repos is passed to the backend. However, the backend does not keep it empty; it just splits text from the blog link and creates an invalid repo URL.

Let me be clear, "Does every blog post highlight need to be linked to a repo?"

CC: @BekahHW @bdougie

Comment (BekahHW) [reply components/molecules/HighlightInput/highlight-input-form.tsx]: Ok, so you're saying from a backend perspective, we _have_ to have a repo listed?

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**